src/test/java/uk/gov/hmcts/reform/ccd/data/CaseDataRepositoryTest.java

```diff
@@ -0,0 +1,85 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.ccd.config.TestConfig;
+import uk.gov.hmcts.reform.ccd.data.entity.CaseDataEntity;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.ccd.fixture.TestData.DELETABLE_CASE_TYPE;
+import static uk.gov.hmcts.reform.ccd.fixture.TestData.JURISDICTION;
+import static uk.gov.hmcts.reform.ccd.fixture.TestData.YESTERDAY;
+
+@DataJpaTest
+@ExtendWith(SpringExtension.class)
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@EnableJpaRepositories(basePackages = {"uk.gov.hmcts.reform.ccd.data"})
+@EntityScan(basePackages = {"uk.gov.hmcts.reform.ccd.data.entity"})
+@ActiveProfiles("unittest")
+@TestPropertySource(properties = {
+    "spring.jpa.hibernate.ddl-auto=create-drop",
+    "spring.liquibase.enabled=false",
+    "spring.flyway.enabled=true"
+})
+@ContextConfiguration(classes = {TestConfig.class})
+public class CaseDataRepositoryTest {
+
+    @Autowired
+    private CaseDataRepository caseDataRepository;
+
+    @BeforeEach
+    public void setUp() {
+        caseDataRepository.deleteAll();
+        CaseDataEntity caseDataEntity = new CaseDataEntity();
+        caseDataRepository.save(getCaseDataEntity(1L));
+        caseDataRepository.save(getCaseDataEntity(2L));
+    }
+
+    @Test
+    void testFindExpiredCases() {
+        List<CaseDataEntity> caseDataAll = caseDataRepository.findAll();
+        assertThat(caseDataAll).hasSize(2);
+        List<CaseDataEntity> caseData = caseDataRepository.findExpiredCases(List.of(DELETABLE_CASE_TYPE));
+        assertThat(caseData).hasSize(2);
+    }
+
+    @Test
+    void testFindByCaseReference() {
+        Optional<CaseDataEntity> caseData = caseDataRepository.findByReference(1L);
+        assertThat(caseData).isPresent();
+        assertThat(caseData.get().getReference()).isEqualTo(1L);
+    }
+
+    @Test
+    void testDeleteCaseData() {
+        Optional<CaseDataEntity> caseData = caseDataRepository.findByReference(1L);
+        assertThat(caseData).isPresent();
+        caseDataRepository.delete(caseData.get());
+        List<CaseDataEntity> caseDataEntities = caseDataRepository.findAll();
+        assertThat(caseDataEntities).hasSize(1);
```

Comment:
Please assert that the particular case you deleted is actually deleted

Reply:
It's now checking case has been deleted